### PR TITLE
Problem: no integration test case for fee calculation of reverted contract call tx.

### LINKED ISSUE #294: Problem: no integration test case for fee calculation of reverted contract call tx.
add regression test case for fee calculation of reverted contract call tx.

### PR DESCRIPTION
**Pull request by macong-cdc**
closes #294 

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: integration_tests/test_basic.py [hunk 1]
```diff
@@ -334,10 +334,10 @@ def test_transaction(cronos):
     with concurrent.futures.ThreadPoolExecutor(4) as executor:
         futures = []
         futures.append(
-            executor.submit(contracts["test_revert_1"].transfer, 5 * (10**18) - 1)
+            executor.submit(contracts["test_revert_1"].transfer, 5 * (10 ** 18) - 1)
```
Comment (yihuang) [inline integration_tests/test_basic.py:337]: You need to update the `black` version to latest, otherwise, it'll fail the one in CI.

Comment (macong-cdc) [reply integration_tests/test_basic.py:337]: Do I change it from here?
https://github.com/macong-cdc/cronos/blob/test-revert-contract-call/integration_tests/pyproject.toml#L12

Or should I change it at somewhere else?

File: integration_tests/test_basic.py [hunk 2]
```diff
@@ -690,3 +690,34 @@ def test_tx_inclusion(cronos, max_gas_wanted):
...
+    tx = the_revert_contract.functions.trigger_revert(dummy).buildTransaction(
+        {"gas": more_than_enough_gas}
+    )
+
+    balance_bef = w3.eth.get_balance(ADDRS["community"])
+    tx_receipt = send_transaction(w3, tx, KEYS["community"])
+    balance_aft = w3.eth.get_balance(ADDRS["community"])
+
+    assert tx_receipt["status"] == 0, "should be failed tx"
+    assert tx_receipt["gasUsed"] != more_than_enough_gas
```
Comment (yihuang) [inline integration_tests/test_basic.py:719]: I think this check will success no matter the contract reverted or not?

File: integration_tests/test_basic.py (hunk 1 above)
Comment (yihuang) [reply integration_tests/test_basic.py:337]: I mean your dev env

Comment (yihuang) [reply integration_tests/test_basic.py:337]: The repo is using 22.6 already: https://github.com/macong-cdc/cronos/blob/test-revert-contract-call/integration_tests/poetry.lock#L91

**Review by yihuang — APPROVED** [review]
LGTM, thanks

**Comment by thomas-nguy** [comment]
what is the purpose of this test?

Is it for the refund logic?

File: integration_tests/test_basic.py (hunk 2 above)
Comment (thomas-nguy) [inline integration_tests/test_basic.py:719]: I think we can test that the gasUsed is below the treshold Maxgax/2

**Comment by macong-cdc** [comment]
> what is the purpose of this test?
> 
> Is it for the refund logic?

Yes, it is for the refund logic for this [issue](https://github.com/evmos/ethermint/pull/751).

File: integration_tests/test_basic.py [hunk 3]
```diff
@@ -401,10 +401,25 @@ def test_exception(cluster):
         w3,
         CONTRACTS["TestRevert"],
     )
-    with pytest.raises(web3.exceptions.ContractLogicError):
-        send_transaction(
-            w3, contract.functions.transfer(5 * (10**18) - 1).buildTransaction()
-        )
```
Comment (yihuang) [inline integration_tests/test_basic.py:407]: I think we can keep this one.

**Pull request merged**